contracts/mirror/src/contract.rs

```diff
@@ -93,37 +89,65 @@ pub fn execute(
             Ok(Response::new())
         }
-        ExecuteMsg::Complete { items } => execute_complete(deps, env, info, items),
-        ExecuteMsg::ChangeReturnType { id, return_type } => {
-            execute_change_return_type(deps, env, info, id, return_type)
-        }
-        ExecuteMsg::UpdateBond {
-            id,
-            receiver,
-            backup,
-            return_type,
-        } => execute_update_bond(deps, env, info, id, receiver, backup, return_type),
         ExecuteMsg::UpdateConfig { new_config } => execute_update_config(deps, info, new_config),
+        ExecuteMsg::Retry { receiver } => execute_retry(deps, env, receiver),
     }
 }
 
-pub fn execute_update_config(
+fn execute_retry(
+    deps: DepsMut<NeutronQuery>,
+    env: Env,
+    receiver: String,
+) -> ContractResult<Response<NeutronMsg>> {
+    let failed_transfers = FAILED_TRANSFERS.may_load(deps.storage, receiver.clone())?;
+    let Config {
+        source_port,
+        source_channel,
+        ibc_timeout,
+        ..
+    } = CONFIG.load(deps.storage)?;
+    let mut ibc_transfer_msgs: Vec<CosmosMsg<NeutronMsg>> = vec![];
+    let mut attrs: Vec<Attribute> = vec![attr("action", "execute_retry")];
+    if let Some(failed_transfers) = failed_transfers {
+        for coin in failed_transfers.iter() {
+            ibc_transfer_msgs.push(CosmosMsg::Custom(NeutronMsg::IbcTransfer {
+                source_port: source_port.clone(),
+                source_channel: source_channel.clone(),
+                token: coin.clone(),
+                sender: env.contract.address.to_string(),
+                receiver: receiver.clone(),
+                timeout_height: RequestPacketTimeoutHeight {
+                    revision_number: None,
+                    revision_height: None,
+                },
+                timeout_timestamp: env.block.time.plus_seconds(ibc_timeout).nanos(),
+                memo: "".to_string(),
+                fee: query_ibc_fee(deps.as_ref(), LOCAL_DENOM)?,
+            }));
+            attrs.push(attr("receiver", receiver.clone()));
+            attrs.push(attr("amount", coin.to_string()));
+        }
+    }
+    // During the IBC transfers we need to remove these funds from state so we can't call retry again for the same user
+    // If any IBC transaction fails then we restore failed transfers for given user in sudo-error
+    // It doesn't throw any exception if given key doesn't exist
+    FAILED_TRANSFERS.remove(deps.storage, receiver);
+    Ok(response("retry", CONTRACT_NAME, attrs).add_messages(ibc_transfer_msgs))
+}
+
+fn execute_update_config(
     deps: DepsMut<NeutronQuery>,
     info: MessageInfo,
     new_config: ConfigOptional,
 ) -> ContractResult<Response<NeutronMsg>> {
     cw_ownable::assert_owner(deps.storage, &info.sender)?;
     let mut config = CONFIG.load(deps.storage)?;
-    let mut attrs = vec![attr("action", "update_config")];
+    let mut attrs = vec![attr("action", "execute-update_config")];
```

Comment:
Please be consistent in the action naming convention
```suggestion
    let mut attrs = vec![attr("action", "execute_update_config")];
```